Item12849: Add Wildcard ACL support in support of deprecating blank DENY ACLs

This patch enables ALLOW and DENY ACLs to use the "*" wildcard to match any user

~~~PATCH 255dd737960a3b0acf5490ef0c924dcc:3f71e0f6781177d4ab610e6445db6c42 lib/Foswiki/Users.pm (Foswiki-1.1.7)
--- lib/Foswiki/Users.pm	2012-12-01 22:33:43.000000000 -0500
+++ lib/Foswiki/Users.pm	2014-04-11 20:08:42.785200044 -0400
@@ -645,6 +645,11 @@
 
     foreach my $ident (@$userlist) {
 
+        # The Wildcard match.  Any user matches the "*" identifier.
+        if ( $ident eq '*' ) {
+            return 1;
+        }
+
         my $identCUID = $this->getCanonicalUserID($ident);
 
         if ( defined $identCUID ) {

